test/metabase/cmd/compare_h2_dbs.clj

```diff
@@ -41,11 +43,16 @@
 (defn- table-names
   "Return a sorted collection of all non-system table names."
   [spec]
-  (jdbc/with-db-metadata [metadata spec]
-    (let [result (jdbc/metadata-result
-                  (.getTables metadata nil "PUBLIC" nil
-                              (into-array String ["TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW"])))]
-      (sort (remove ignored-table-names (map :table_name result))))))
+  (sql-jdbc.execute/do-with-connection-with-options
+   :h2
+   spec
+   nil
+   (fn [^java.sql.Connection conn]
+     (let [metadata (.getMetaData conn)
+           result (jdbc/metadata-result
+                   (.getTables metadata nil "PUBLIC" nil
+                               (into-array String ["TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW"])))]
+       (sort (remove ignored-table-names (map :table_name result)))))))
```

Comment:
Fixes this linting warning:

https://github.com/metabase/metabase/blob/354f152e113724e50a513a64db118fc14bc8684f/.clj-kondo/config.edn#L135